### PR TITLE
YLP-2476: Adapt blip to use the new route `/all` on tideV2

### PR DESCRIPTION
**Pull request by ydrisR**
## Checklist
- [x] PR name contains the Jira ticket number
- [x] Changelog updated
- [x] Code tested (unit tests and integration tests)
- [x] No console errors/warnings in integration tests
- [x] No translation files updated (except for English files)
- [x] Code formatted on every file
- [x] Copyright added on every file
- [x] End-to-end tests successful on CI

### CODE REVIEW

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:269]: We usually avoid having "console.log" (it often pollute the console with information that does not feel important). In this case, I think that it is better to use "console.error'
```suggestion
        console.error({ message, rawData: raw })
```

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:275]: How comes that we have to check that the value is defined? Doesn't the API returns an empty array instead of null?

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:377]: Is there a JIRA ticket linked to this "TODO"? IMO it is better to have the JIRA ticket and no TODO (you end up finding "TODO" two years later in the code and they are still not done)

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (TimCanu) [inline packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:349]: This whole function is tricky but as you didn't write it, I'm ok with this :slightly_smiling_face:

File: packages/yourloops/test/integration/assert/patient-statistics.assert.ts
Comment (TimCanu) [inline packages/yourloops/test/integration/assert/patient-statistics.assert.ts:56]: This is suspicious... Why did this value changed?

File: packages/yourloops/test/unit/components/patient-data/patient-data.utils.test.ts
Comment (TimCanu) [inline packages/yourloops/test/unit/components/patient-data/patient-data.utils.test.ts:76]: Questionnable yes, it does bring little value...

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (ydrisR) [reply packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:275]: To be on the safe side ;)

File: package-lock.json
Comment (celia-rouis) [inline package-lock.json:7396]: **❔ question:** ‏Was this change wanted? Or is it an issue following a merge conflict? I find it weird to be retrieving the package directly from the NPM registry instead of Nexus

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (celia-rouis) [inline packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:232]: **🤓 nitpick:** ‏So sad to have all `if()` conditions sorted in alphabetical order, to find `data.alarmEvents` at the end... :joy:

File: packages/yourloops/components/dashboard-widgets/device-usage-widget.tsx
Comment (celia-rouis) [inline packages/yourloops/components/dashboard-widgets/device-usage-widget.tsx:96]: **💭 thought:** ‏Just an idea because this line of code makes me think of it: if we know that we should always have one element in the `medicalData.pumpSettings` array, could the API send us only this element instead of an array? It would allow us to avoid the computation of the last pump settings every time we need it

File: packages/yourloops/components/patient-data/patient-data.utils.ts
Comment (celia-rouis) [inline packages/yourloops/components/patient-data/patient-data.utils.ts:169]: **🤓 nitpick:** ‏Almost the alphabetical order... :sweat_smile:

File: packages/yourloops/test/integration/mock/data.api.mock.ts
Comment (celia-rouis) [inline packages/yourloops/test/integration/mock/data.api.mock.ts:98]: **🎯 suggestion (non-blocking):** It looks like the methods `convertCbgMg2Mmol()` and `convertSmbgMg2Mmol()` are doing the same thing, could they be factorized?‏

File: packages/yourloops/test/unit/components/patient-data/data.mock.ts
Comment (celia-rouis) [inline packages/yourloops/test/unit/components/patient-data/data.mock.ts:28]: **🤓 nitpick:** ‏
```suggestion
export const mockDataFromApi = {
```

**Review by TimCanu — COMMENTED** [review]
Very nice :slightly_smiling_face: 
I still need to test it but otherwise I have no comments

File: packages/yourloops/components/dashboard-widgets/device-usage-widget.tsx
Comment (ydrisR) [reply packages/yourloops/components/dashboard-widgets/device-usage-widget.tsx:96]: Agree, I created a ticket to work on this later (next release)
https://diabeloop.atlassian.net/browse/YLP-2695

File: packages/yourloops/components/patient-data/patient-data.utils.ts
Comment (ydrisR) [reply packages/yourloops/components/patient-data/patient-data.utils.ts:169]: ok

File: packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts
Comment (ydrisR) [reply packages/medical-domain/src/domains/repositories/medical/medical-data.service.ts:232]: ok

File: packages/yourloops/test/integration/mock/data.api.mock.ts
Comment (ydrisR) [reply packages/yourloops/test/integration/mock/data.api.mock.ts:105]: corrected

File: packages/yourloops/components/device/pump-info-table.tsx
Comment (TimCanu) [inline packages/yourloops/components/device/pump-info-table.tsx:56]:
```suggestion
              {pump.manufacturer ? pump.manufacturer : t('N/A')}</Typography>
```

File: packages/yourloops/components/device/pump-info-table.tsx
Comment (TimCanu) [inline packages/yourloops/components/device/pump-info-table.tsx:64]:
```suggestion
            <Typography variant="body2" className="bold">{pump.serialNumber ? pump.serialNumber : t('N/A')}</Typography>
```

File: packages/yourloops/components/device/pump-info-table.tsx
Comment (TimCanu) [inline packages/yourloops/components/device/pump-info-table.tsx:72]:
```suggestion
            <Typography variant="body2" className="bold">{pump.swVersion ? pump.swVersion : t('N/A')}</Typography>
```

**Review by TimCanu — APPROVED** [review]

**Comment by TimCanu** [comment]
(Obviously do not merge before the yourloops PR is ready to be merged :slightly_smiling_face: )

**Pull request merged**
